src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanRule.java

```diff
@@ -71,6 +79,29 @@ public void init() {
         }
     }
 
+    protected void scan(List<NameValuePair> nameValuePairs) {
+        if (nameValuePairs.isEmpty()
+                || nameValuePairs.get(0).getType() != NameValuePair.TYPE_HEADER) {
+            super.scan(nameValuePairs);
+        } else {
+            nameValuePairs = new ArrayList<>(nameValuePairs);
+            List<HttpHeaderField> headerFields = getBaseMsg().getRequestHeader().getHeaders();
```

Comment:
`getBaseMsg().getRequestHeader().getHeader(HttpRequestHeader.AUTHORIZATION)` and check if not null. More efficient than create the `HttpHeaderField`s and iterate them, but if you prefer keeping the original case for some reason (the `VariantHeader` does not need it) keep it and check the other comment.